src/front/glsl/parser/declarations.rs

```diff
@@ -19,6 +19,55 @@ use crate::{
 
 use super::{DeclarationContext, ParsingContext, Result};
 
+/// Helper method used to retrieve the child type of `ty` at
+/// index `i`.
+///
+/// # Note
+///
+/// Does not check if the index is valid and returns the same type
+/// when indexing out-of-bounds a struct or indexing a non indexable
+/// type.
```

Comment:
You're counting on validation to complain if there are too many components - correct? (It does, in `naga::valid::compose::validate_compose`.)

This PR is still an improvement over the status quo, but tolerating bad indices here does mean that if someone supplies too many values in the initializer, you're going to be passing meaningless types to `implicit_conversion`, and the error messages will be pretty confusing. That is, you'll get weird GLSL front end errors, instead of the nice "too many components" error from the validator, because you'll never make it that far.